zfit/core/parameter.py

```diff
@@ -623,15 +623,19 @@ def __init__(self, params, value_fn, name="BaseComposedParameter", **kwargs):
         if not callable(value_fn):
             raise TypeError("`value_fn` is not callable.")
         n_params = len(signature(value_fn).parameters)
-        # TODO(0.6): remove, legacy?
+        # TODO(0.6): change, remove legacy?
         if n_params == 0:
-            warnings.warn("The `value_fn` for composed parameters should take the same number"
-                          " of arguments as `params` are given.")
-            legacy_value_fn = value_fn
-
-            def value_fn(*_):
-                return legacy_value_fn()
-        # end legacy
+            if len(params) == 0:
+                warnings.warn("No `params` specified, the `value_fn` is supposed to return a constant. "
```

Comment:
This warning might also make sense if `n_params > 0`. Maybe add a test to see if `n_params == len(params)`.
Also naming is a bit confusing here due to the double meaning of params. Maybe call it `n_function_parameters` instead of `n_params`.

Reply:
I did not add the test on purpose, as there can be variable arguments lengths in the signature. Renaming `n_params` makes sense